spring-boot-modules/spring-boot-validations/src/test/java/com/baeldung/controller/ValidationControllerUnitTest.java

```diff
@@ -0,0 +1,112 @@
+    static class EmployeeServiceImplTestContextConfiguration {
+        @Bean
+        public ValidationService validationService() {
+            return new ValidationService() {
+            };
+        }
+    }
+
+    @Autowired
+    ValidationService service;
+
+    @Test
+    void whenNullInputForBooleanField_thenHttpBadRequestAsHttpResponse() throws Exception {
+
+        String postBody = "{\"boolField\":null,\"trueField\":true,\"falseField\":false,\"boolStringVar\":\"+\"}";
+
+        mockMvc.perform(post("/validateBoolean").contentType("application/json")
+            .content(postBody))
+            .andExpect(status().isBadRequest());
+
+    }
+
+    @Test
+    void whenInvalidInputForTrueBooleanField_thenErrorResponse() throws Exception {
+
+        String postBody = "{\"boolField\":true,\"trueField\":false,\"falseField\":false,\"boolStringVar\":\"+\"}";
+
+        String output = mockMvc.perform(post("/validateBoolean").contentType("application/json")
+            .content(postBody))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+        assertEquals("trueField must have true value", output);
+    }
+
+    @Test
+    void whenInvalidInputForFalseBooleanField_thenErrorResponse() throws Exception {
+
+        String postBody = "{\"boolField\":true,\"trueField\":true,\"falseField\":true,\"boolStringVar\":\"+\"}";
+
+        String output = mockMvc.perform(post("/validateBoolean").contentType("application/json")
+            .content(postBody))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+        assertEquals("falseField must have false value", output);
+    }
+
+    @Test
+    void whenInvalidBooleanFromJson_thenErrorResponse() throws Exception {
+
+        String postBody = "{\"boolField\":true,\"trueField\":true,\"falseField\":false,\"boolStringVar\":\"plus\"}";
+
+        String output = mockMvc.perform(post("/validateBoolean").contentType("application/json")
+            .content(postBody))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+        assertEquals("Only values accepted as Boolean are + and -", output);
+
+    }
+
+    @Test
+    void whenAllBooleanFieldsValid_thenCorrectResponse() throws Exception {
+
+        String postBody = "{\"boolField\":true,\"trueField\":true,\"falseField\":false,\"boolStringVar\":\"+\"}";
+
+        String output = mockMvc.perform(post("/validateBoolean").contentType("application/json")
+            .content(postBody))
+            .andReturn()
+            .getResponse()
+            .getContentAsString();
+
+        assertEquals("BooleanObject is valid", output);
+
+    }
+
+    @Test
+    void givenAllBooleanFieldsValid_whenServiceValidationFails_thenErrorResponse() throws Exception {
+
+        mockMvc.perform(post("/validateBooleanAtService").contentType("application/json"))
+            .andExpect(status().isInternalServerError());
+    }
```

Comment:
Add these two tests:

@Test
    void whenNullInputForTrueBooleanField_thenCorrectResponse() throws Exception {

        String postBody = "{\"boolField\":true,\"trueField\":null,\"falseField\":false,\"boolStringVar\":\"+\"}";

        mockMvc.perform(post("/validateBoolean").contentType("application/json")
                        .content(postBody))
                .andExpect(status().isOk());

    }

    @Test
    void whenNullInputForFalseBooleanField_thenHttpBadRequestAsHttpResponse() throws Exception {
    
        String postBody = "{\"boolField\":true,\"trueField\":true,\"falseField\":null,\"boolStringVar\":\"+\"}";
    
        mockMvc.perform(post("/validateBoolean").contentType("application/json")
                        .content(postBody))
                .andExpect(status().isBadRequest());
    
    }

Reply:
Test Case added. I have just tweaked the last test case to assert with relevant error message instead of status code.